internal/nodes/avi_model_evh_nodes.go

```diff
@@ -1484,6 +1493,16 @@ func buildWithInfraSettingForEvh(key string, vs *AviEvhVsNode, vsvip *AviVSVIPNo
 			vs.EnableRhi = &enableRhi
 		}
 
+		if vs.EnableRhi != nil && *vs.EnableRhi {
+			if infraSetting.Spec.Network.BgpPeerLabels != nil {
+				vsvip.BGPPeerLabels = &infraSetting.Spec.Network.BgpPeerLabels
+			} else {
+				vsvip.BGPPeerLabels = lib.GetBgpPeerLabels()
```

Comment:
I wonder if this is the right behavior. What if the user just wanted the RHI to be true but they didn't want to specify a BGP peer label in the infra setting because they don't want this feature for this IngressClass/AviInfraCRD. Shouldn't we just have enable RHI with no BGP peer labels in that case?

Reply:
Yes that's covered as part of the  conditional i believe. If enableRhi is set to true, then the bgp peer labels are set, from either the infra CRD, or the config map. if not set then the labels are not added/removed.